host/4/bookworm/python/python312/python312-composite.template

```diff
@@ -0,0 +1,71 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+FROM mcr.microsoft.com/oryx/python:3.12-debian-bookworm as python
+
+
+# Install Python dependencies
+# MS SQL related packages: unixodbc msodbcsql17 mssql-tools
+# .NET Core dependencies: --no-install-recommends ca-certificates libc6 libgcc1 libgssapi-krb5-2 libicu67 libssl1.1 libstdc++6 zlib1g
+# OpenCV dependencies:libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb
+# binutils: binutils
+# OpenMP dependencies: libgomp1 && \
+# Fix from https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157 : libc-dev
+# mysql dependencies: default-libmysqlclient-dev
+# Azure ML dependencies: liblttng-ust0
+RUN apt-get update && \
+    apt-get install -y wget apt-transport-https curl gnupg locales && \
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    curl https://packages.microsoft.com/keys/microsoft.asc | tee /etc/apt/trusted.gpg.d/microsoft.asc && \
+    curl https://packages.microsoft.com/config/debian/12/prod.list | tee /etc/apt/sources.list.d/mssql-release.list && \
+    # Needed for libss1.0.0 and in turn MS SQL
+    echo 'deb http://security.debian.org/debian-security bookworm-security main' >> /etc/apt/sources.list && \
+    # install MS SQL related packages.pinned version in PR # 1012.
+    echo 'en_US.UTF-8 UTF-8' > /etc/locale.gen && \
+    locale-gen && \
+    apt-get update && \
+    ACCEPT_EULA=Y apt-get install -y unixodbc msodbcsql18=18.2.2.1-1 mssql-tools18 &&\
+    apt-get install -y --no-install-recommends ca-certificates \
+    libc6 libgcc1 libgssapi-krb5-2 libicu67 libssl1.1 libstdc++6 zlib1g &&\
+    apt-get install -y libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb binutils\
+    binutils libgomp1 default-libmysqlclient-dev liblttng-ust0 && \
+    rm -rf /var/lib/apt/lists/*
+
+FROM mcr.microsoft.com/dotnet/runtime-deps:6.0
+ARG HOST_VERSION
+
+COPY --from=runtime-image [ "/FuncExtensionBundles", "/FuncExtensionBundles" ]
+COPY sshd_config /etc/ssh/
+COPY start.sh /azure-functions-host/
+COPY install_ca_certificates.sh /opt/startup/
+COPY --from=runtime-image ["/azure-functions-host", "/azure-functions-host"]
+COPY --from=runtime-image [ "/workers/python/3.12/LINUX", "/azure-functions-host/workers/python/3.12/LINUX" ]
+COPY --from=runtime-image [ "/workers/python/worker.config.json", "/azure-functions-host/workers/python" ]
+COPY --from=python [ "/", "/" ]
```

Comment:
This line is causing the issue. Basically when you copy the whole of the content of oryx to this image. You overwrite the apt-key location that stored the pubkey necessary for this package feed. If you were able to stop the build at this step you would see that your image is missing the erroring key in the following directory: 
![image](https://github.com/Azure/azure-functions-docker/assets/46761504/36c70d54-2671-444a-8d85-f2b5b64b1002)

Reply:
I was running the python312.dockerfile. In that file, we file do an apt-get update and then run these copy commands.

Reply:
Could you expand on this I don't understand what you mean

Reply:
In python312.Dockerfile, the error was thrown at line 21 when running an apt-get update..
The COPY statements you mentioned in the comments are in line 83

Reply:
Fixed this error.